### PR TITLE
Special case fetching for driver tests

### LINKED ISSUE #107: pprof tests leave orphaned files in $HOME/pprof
These should either not be generated, or generated in a test directory and cleaned up.

### PR DESCRIPTION
**Pull request by rauls5382**
Use a more regular mechanism to name sources during driver tests and
recognize the naming during fetching to avoid saving the test profiles
into $HOME/pprof.

Also move the https+insecure testing into fetch_test, to focus it on
fetching the profile.

This fixes #107

### CODE REVIEW

File: internal/driver/driver_test.go [hunk 1]
```diff
@@ -227,9 +218,13 @@ func addFlags(f *testFlags, flags []string) {
 	}
 }
 
+func testSourceUrl(port int) string {
+	return fmt.Sprintf("http://%s:%d/", testSourceAddress, port)
```
Comment (tamird) [inline internal/driver/driver_test.go]: consider using `net.JoinHostPort` here; this bites people in subtle ways.

File: internal/driver/driver_test.go [hunk 2]
```diff
@@ -437,17 +436,9 @@ func testFetchSymbols(source, post string) ([]byte, error) {
 			fmt.Fprintf(&buf, "%#x\n", a-testStart-testOffset)
 		}
 		return buf.Bytes(), nil
+	default:
+		return nil, fmt.Errorf("unexpected source :%s", source)
```
Comment (tamird) [inline internal/driver/driver_test.go]: misplaced colon

Comment (rauls5382) [inline internal/driver/driver_test.go]: done, thx.

File: internal/driver/driver_test.go [hunk 3]
```diff
@@ -227,9 +218,13 @@ func addFlags(f *testFlags, flags []string) {
 	}
 }
 
+func testSourceURL(port int) string {
+	return fmt.Sprintf("http://%s:%d/", testSourceAddress, port)
```
Comment (tamird) [inline internal/driver/driver_test.go]: consider using `net.JoinHostPort` here; this bites people in subtle ways. (previous comment was eaten by golint commit).

**Comment by aalexand** [comment]
Remove emptyFlags, unused now.

**Comment by tamird** [comment]
Just FYI: removing the unused code appeases `unused`, not `golint` :).

**Comment by aalexand** [comment]
@rauls5382 rebase & resolve the conflicts?

**Comment by rauls5382** [comment]
I give up -- will send this on a separate CL

**Comment by aalexand** [comment]
FWIW, you could probably recover with

    git stash # to make sure no local changes
    git reset --hard HEAD^ # to discard the merge you did
    git rebase -i master # to rebase interactively resolving the conflicts

**Comment by aalexand** [comment]
Plus you'd have to push the branch with --force to overwrite the history.

**Pull request closed without merging**